enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,209 @@
+---
+title: optionally-deploy-csi-plugin
+authors:
+  - "@copejon"
+reviewers:
+  - "@pacevedom: MicroShift team-lead"
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+  - "@jakobmoellerdev Edge Enablement LVMO SME"
+  - "@suleymanakbas91,  Edge Enablement LVMO SME"
+approvers:
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+api-approvers:
+  - "None"
+creation-date: 2024-06-10
+last-updated: 2024-06-11
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-856
+see-also:
+  - "enhancements/microshift/microshift-default-csi-plugin.md"
+---
+
+# Optionally Deploy CSI Plugin
+
+## Summary
+
+MicroShift is a small form-factor, single-node OpenShift targeting IoT and Edge Computing use cases characterized by
+tight resource constraints, unpredictable network connectivity, and single-tenant workloads. See
+[kubernetes-for-devices-edge.md](./kubernetes-for-device-edge.md) for more detail.
+
+Out of the box, MicroShift includes the LVMS CSI driver and CSI snapshotter. The LVM Operator is not itself packaged
+with the platform, in adherence with the project's [guiding principles](./kubernetes-for-device-edge.md#goals). See the
+[MicroShift Default CSI Plugin](microshift-default-csi-plugin.md) proposal for a more in-depth explanation of the
+storage provider and reasons for its integration into MicroShift. Configuration of the driver is exposed via a config
+file at /etc/microshift/lvmd.yaml. The manifests required to deploy and configure the driver and snapshotter are baked
+into the MicroShift binary during compilation and are deployed during runtime. This means that even if a user wanted to
+disable the driver and/or snapshotter, they would be deployed again when the service restarts. Additionally, the LVMS an
+CSI containers required to run LVMS are always packaged with the product and thus will always consume a certain amount
+of storage space devices where MicroShift is deployed.
+
+## Motivation
+
+Not all users may persistent storage provider or volume snapshot capabilities for their MicroShift deployments and
+should be enabled to choose whether to deploy the driver and / or snapshotter. This will afford resource-conscious users
```

Comment:
This is already possible today through the "enabled" option, I dont really see why this is not possible today.

Reply:
MicroShift doesn't have a config option to enable/disable CSI. Can you point me to what your referring to?

Reply:
I am referring to the fact that it is quite easily possible to build a flag or something similar into the config that disables the driver. I guess Im trying to convey that we dont need such a major rework for getting that done. There may be other reasons for the extraction such as modularity of the CSI driver of course.

Reply:
Using a config field doesn't solve the goal of separating the LVMS project from microshift. It also poses a new set of complexities where as revoking LVMS's first-class citizen status in MicroShift would provide a much more simplified solution and reduce tech debt and testing requirements.